lib/metasploit/framework/login_scanner/redis.rb

```diff
@@ -0,0 +1,89 @@
+require 'metasploit/framework/login_scanner/base'
+require 'metasploit/framework/login_scanner/rex_socket'
+require 'metasploit/framework/tcp/client'
+
+module Metasploit
+  module Framework
+    module LoginScanner
+
+      # This is the LoginScanner class for dealing with REDIS.
+      # It is responsible for taking a single target, and a list of credentials
+      # and attempting them. It then saves the results.
+
+      class REDIS
+        include Metasploit::Framework::LoginScanner::Base
+        include Metasploit::Framework::LoginScanner::RexSocket
+        include Metasploit::Framework::Tcp::Client
+
+        DEFAULT_PORT         = 6379
+        LIKELY_PORTS         = [ DEFAULT_PORT ]
+        LIKELY_SERVICE_NAMES = [ 'redis' ]
+        PRIVATE_TYPES        = [ :password ]
+        REALM_KEY            = nil
+
+        # This method can create redis command which can be read by redis server
+        def redis_proto(command_parts)
+          return if command_parts.blank?
+          command = "*#{command_parts.length}\r\n"
+          command_parts.each do |p|
+            command << "$#{p.length}\r\n#{p}\r\n"
+          end
+          command
+        end
+
+        # This method attempts a single login with a single credential against the target
+        # @param credential [Credential] The credential object to attempt to login with
+        # @return [Metasploit::Framework::LoginScanner::Result] The LoginScanner Result object
+        def attempt_login(credential)
+          result_options = {
+            credential: credential,
+            status: Metasploit::Model::Login::Status::INCORRECT,
+            host: host,
+            port: port,
+            protocol: 'tcp',
+            service_name: 'redis'
+          }
+
+          disconnect if self.sock
+
+          begin
+            connect
+            select([sock], nil, nil, 0.4)
+
+            command = redis_proto(['PING'])
+            sock.put(command)
+            result_options[:proof] = sock.get_once
+
+            if result_options[:proof] && result_options[:proof] =~ /(?<auth_response>ERR operation not permitted|NOAUTH Authentication required)/i
```

Comment:
No need to do a named capture with `auth_response` since you are not using it afterwards.  

Also, there is a problem with assuming that `ERR operation not permitted` means that authentication is required, although I do realize that at least some versions do work this way.  I believe this same message will also occur if the particular command is remapped on the redis server side, which is an option available by default in the redis configuration files I've seen.  Perhaps it should be refactored to attempt authentication any time it gets any response that looks like it came from redis.  

That also raises the issue of how this code handles the case when redis authentication is not required.  It doesn't look like that is handled and I suspect what might happen is that it will mark every credential as valid, or only the first one attempted, or something like that, none of which are ideal.  It should uniquely identify this situation and indicate that no credentials are required.

Lastly, you may want to make the command tried configurable as I did in the other redis module(s).  This will help users of the module work around situations where `PING` is remapped.